actors/user/src/main/java/org/sunbird/user/actors/UserProfileReadActor.java

```diff
@@ -587,102 +593,138 @@ private void getUserDetailsByLoginId(Request actorMessage) {
-                ResponseCode.SERVER_ERROR.getResponseCode());
-        sender().tell(exception, self());
-        return;
+  private void getUserByKey(Request actorMessage) {
+    String key = (String) actorMessage.getRequest().get(JsonKey.KEY);
+    String value = (String) actorMessage.getRequest().get(JsonKey.VALUE);
+    String encryptedValue = null;
+    try {
+      encryptedValue = encryptionService.encryptData(value);
+    } catch (Exception e) {
+      ProjectCommonException exception =
+          new ProjectCommonException(
+              ResponseCode.userDataEncryptionError.getErrorCode(),
+              ResponseCode.userDataEncryptionError.getErrorMessage(),
+              ResponseCode.SERVER_ERROR.getResponseCode());
+      sender().tell(exception, self());
+      return;
+    }
+    UserDao userDao = new UserDaoImpl();
+    Map<String, Object> searchMap = new HashMap();
+    searchMap.put(key, encryptedValue);
+    List<User> foundUsers = userDao.getUsersByProperties(searchMap);
+    if (foundUsers == null || foundUsers.size() == 0) {
+      throw new ProjectCommonException(
+          ResponseCode.userNotFound.getErrorCode(),
+          ResponseCode.userNotFound.getErrorMessage(),
+          ResponseCode.RESOURCE_NOT_FOUND.getResponseCode());
+    }
+    User foundUser = foundUsers.get(0);
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    Map<String, Object> result = objectMapper.convertValue(foundUser, Map.class);
+    sendResponse(actorMessage, result);
+  }
+
+  private void sendResponse(Request actorMessage, Map<String, Object> result) {
+    if (result == null || result.size() == 0) {
+      throw new ProjectCommonException(
+          ResponseCode.userNotFound.getErrorCode(),
+          ResponseCode.userNotFound.getErrorMessage(),
+          ResponseCode.RESOURCE_NOT_FOUND.getResponseCode());
+    }
+
+    // check whether is_deletd true or false
+    if (ProjectUtil.isNotNull(result)
+        && result.containsKey(JsonKey.IS_DELETED)
+        && ProjectUtil.isNotNull(result.get(JsonKey.IS_DELETED))
+        && (Boolean) result.get(JsonKey.IS_DELETED)) {
+      throw new ProjectCommonException(
+          ResponseCode.userAccountlocked.getErrorCode(),
+          ResponseCode.userAccountlocked.getErrorMessage(),
+          ResponseCode.CLIENT_ERROR.getResponseCode());
+    }
+    fetchRootAndRegisterOrganisation(result);
+    // having check for removing private filed from user , if call user and response
+    // user data id is not same.
+    String requestedById =
+        (String) actorMessage.getContext().getOrDefault(JsonKey.REQUESTED_BY, "");
+    ProjectLogger.log(
+        "requested By and requested user id == "
+            + requestedById
+            + "  "
+            + (String) result.get(JsonKey.USER_ID));
+
+    try {
+      if (!(((String) result.get(JsonKey.USER_ID)).equalsIgnoreCase(requestedById))) {
+        result = removeUserPrivateField(result);
```

Comment:
We need to provide user name as well , so remove  userName key from  projectUtil :   String[] excludes (line 65)

Reply:
From System setting , in get userProfile v2 we are removing username , after reading system setting , that also need to be updated.